rio_toa/reflectance.py

```diff
@@ -84,37 +115,72 @@ def calculate_landsat_reflectance(src_path, src_mtl, dst_path, creation_options,
     """
     mtl = toa_utils._load_mtl(src_mtl)
 
-    M = toa_utils._load_mtl_key(mtl,
-        ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_MULT_BAND_'],
-        band)
-    A = toa_utils._load_mtl_key(mtl,
-        ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_ADD_BAND_'],
-        band)
-    E = toa_utils._load_mtl_key(mtl, 
-        ['L1_METADATA_FILE', 'IMAGE_ATTRIBUTES','SUN_ELEVATION'])
+    E = mtl['L1_METADATA_FILE']['IMAGE_ATTRIBUTES']['SUN_ELEVATION']
+    date_collected = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['DATE_ACQUIRED']
+    time_collected_utc = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['SCENE_CENTER_TIME']
 
     dst_dtype = np.__dict__[dst_dtype]
+    M, A = [], []
 
-    with rio.open(src_path) as src:
-        dst_profile = src.profile.copy()
+    for band, src_path in zip(bands, src_paths):
+        M.append(toa_utils._load_mtl_key(mtl,
+            ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_MULT_BAND_'],
+            band))
+        A.append(toa_utils._load_mtl_key(mtl,
+            ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_ADD_BAND_'],
+            band))
 
-        src_nodata = src.nodata
+        with rio.open(src_path) as src:
+            dst_profile = src.profile.copy()
+            src_nodata = src.nodata
 
-        for co in creation_options:
-            dst_profile[co] = creation_options[co]
+            for co in creation_options:
+                dst_profile[co] = creation_options[co]
 
-        dst_profile['dtype'] = dst_dtype
+            dst_profile['dtype'] = dst_dtype
 
     global_args = {
         'A': A,
         'M': M,
         'E': E,
         'src_nodata': 0,
-        'dst_dtype': dst_dtype
+        'src_crs': dst_profile['crs'],
+        'dst_dtype': dst_dtype,
+        'rescale_factor': rescale_factor,
+        'pixel_sunangle': pixel_sunangle,
+        'date_collected': date_collected,
+        'time_collected_utc': time_collected_utc,
+        'stack': stack,
+        'bands': len(bands)
     }
 
-    with riomucho.RioMucho([src_path], dst_path, _reflectance_worker,
-        options=dst_profile,
-        global_args=global_args) as rm:
+    if stack:
```

Comment:
@virginiayung shouldn't we just:
 - stack if the number of inputs > 1
 - not stash if it is equal to 1 ?

I can't think of a case where we'd want multiple inputs AND multiple outputs